client/me/purchases/payment-methods/payment-method-edit.tsx

```diff
@@ -0,0 +1,158 @@
+import { ReduxDispatch } from 'calypso/state/redux-store';
+import { editStoredCardTaxLocation } from 'calypso/state/stored-cards/actions';
+import { isEditingStoredCard } from 'calypso/state/stored-cards/selectors';
+import PaymentMethodDetails from './payment-method-details';
+import PaymentMethodEditDialog from './payment-method-edit-dialog';
+
+interface Props {
+	card: PaymentMethod;
+}
+
+const PaymentMethodEdit: FunctionComponent< Props > = ( { card } ) => {
+	const translate = useTranslate();
+	const isEditing = useSelector( ( state ) =>
+		isEditingStoredCard( state, card.stored_details_id )
+	);
+
+	const reduxDispatch = useDispatch< ReduxDispatch >();
+	const [ isDialogVisible, setIsDialogVisible ] = useState( false );
+	const closeDialog = useCallback( () => setIsDialogVisible( false ), [] );
+
+	const renderTaxPostalCode = (): string => {
+		const filtered = card.meta.find(
+			( item: { meta_key: string } ) => item.meta_key === 'tax_postal_code'
+		);
+		return filtered?.meta_value ?? '';
+	};
+
+	const renderTaxCountryCode = (): string => {
+		const filtered = card.meta.find(
+			( item: { meta_key: string } ) => item.meta_key === 'tax_country_code'
+		);
+		return filtered?.meta_value ?? '';
+	};
+
+	const [ inputValues, setInputValues ] = useState( {
+		tax_postal_code: '',
+		tax_country_code: '',
+	} );
+
+	const handleEdit = useCallback( () => {
+		closeDialog();
+
+		reduxDispatch(
+			editStoredCardTaxLocation( card, inputValues.tax_postal_code, inputValues.tax_country_code )
+		)
+			.then( () => {
+				if ( isPaymentAgreement( card ) ) {
+					reduxDispatch( successNotice( translate( 'Payment method edited successfully' ) ) );
+				} else {
+					reduxDispatch( successNotice( translate( 'Card edited successfully!' ) ) );
+				}
+				recordTracksEvent( 'calypso_purchases_edit_tax_location' );
+			} )
+			.then( () => {
+				window.location.reload();
+			} )
+			.catch( ( error: Error ) => {
+				reduxDispatch( errorNotice( error.message ) );
+			} );
+	}, [
+		closeDialog,
+		card,
+		inputValues.tax_postal_code,
+		inputValues.tax_country_code,
+		translate,
+		reduxDispatch,
+	] );
+
+	const handleSubmit = ( event ) => {
+		event.preventDefault();
+		const { name, value } = event.target;
+		setInputValues( { ...inputValues, [ name ]: value } );
+		handleEdit();
+	};
+
+	const renderEditForm = () => {
+		return (
+			<form onSubmit={ handleSubmit }>
+				<input
+					type="text"
+					name="tax_postal_code"
+					placeholder="Enter postal code"
+					value={ inputValues.tax_postal_code }
+					onChange={ ( e ) =>
+						setInputValues( { ...inputValues, tax_postal_code: e.target.value } )
+					}
+				/>
+				<input
+					type="text"
+					name="tax_country_code"
+					placeholder="Enter country code"
+					value={ inputValues.tax_country_code }
+					onChange={ ( e ) =>
+						setInputValues( { ...inputValues, tax_country_code: e.target.value } )
+					}
+				/>
+			</form>
+		);
+	};
+
+	const renderEditButton = () => {
```

Comment:
Could this function be a component of its own instead? Something that could be used like `{ renderTaxPostalCode() && <EditButton isEditing={ isEditing } onClick={ () => setIsDialogVisible( true ) } \> }`?

Reply:
This was made into a component.